packages/governance/src/types.js

```diff
@@ -179,6 +248,273 @@
+ */
+
+/**
+ * @typedef {Object} GovernorPublic
+ * @property {GovernContract} governContract
+ * @property {() => Instance} getRegistrar
+ * @property {(i: Instance) => boolean} governsContract
+ */
+
+/**
+ * @typedef {Object} ParamSpecification
+ * @property {string} key
+ * @property {string} parameterName
+ */
+
+/**
+ * @typedef {Object} ParamChangeVoteResult
+ * @property {Instance} instance - instance of the BallotCounter
+ * @property {Details} details
+ * @property {Promise<ParamValue>} outcomeOfUpdate - A promise for the result
+ *    of updating the parameter value. Primarily useful for its behavior on
+ *    rejection.
+ */
+
+/**
+ * @typedef {Object} LimitedCreatorFacet
+ *
+ * The creatorFacet for the governed contract that will be passed to the
+ * responsible party. It does not have access to the paramManager.
+ * @property {() => Instance} getContractGovernor
+ */
+
+/**
+ * @typedef {Object} ContractPowerfulCreatorFacet
+ *
+ *   A powerful facet that carries access to both the creatorFacet to be passed
+ *   to the caller and the paramManager, which will be used exclusively by the
+ *   ContractGovenor.
+ * @property {() => ERef<LimitedCreatorFacet>} getLimitedCreatorFacet
+ * @property {() => ParamManagerAccessor} getParamMgrAccessor
+ */
+
+/**
+ * @typedef {Object} GovernedContract
+ * @property {VoteOnParamChange} voteOnParamChange
+ * @property {() => ERef<LimitedCreatorFacet>} getCreatorFacet - creator
+ *   facet of the governed contract, without the tightly held ability to change
+ *   param values.
+ * @property {() => any} getPublicFacet - public facet of the governed contract
+ * @property {() => ERef<Instance>} getInstance - instance of the governed
+ *   contract
+ */
+
+/**
+ * @callback StartGovernedContract
+ * @param {RegistrarCreator} registrarCreatorFacet
+ * @param {Installation} governedContractInstallation
+ * @param {IssuerKeywordRecord} issuerKeywordRecord
+ * @param {Terms} customTerms
+ * @returns {GovernedContract}
+ */
+
+/**
+ * @callback VoteOnParamChange
+ * @param {ParamSpecification} paramSpec
+ * @param {ParamValue} proposedValue
+ * @param {Installation} ballotCounterInstallation
+ * @param {bigint} deadline
+ * @returns {ParamChangeVoteResult}
+ */
+
+/**
+ * @typedef {Object} ParamManagerAccessor
+ * @property {(paramSpec: ParamSpecification) => ParamManagerFull} get
+ */
+
+/**
+ * @typedef {Object} ParamGovernor
+ * @property {VoteOnParamChange} voteOnParamChange
+ * @property {CreatedBallot} createdBallot
+ */
+
+/**
+ * @callback SetupGovernance
+ * @param {ERef<ParamManagerAccessor>} accessor
+ * @param {ERef<PoserFacet>} poserFacet
+ * @param {ERef<Instance>} contractInstance
+ * @param {Timer} timer
+ * @returns {ParamGovernor}
+ */
+
+/**
+ * @callback CreatedBallot
+ *   Was this ballot created by this ContractGovernor?
+ * @param {Instance} ballotInstance
+ * @returns {boolean}
+ */
+
+/**
+ * @callback AssertContractGovernance
+ *
+ * @param {ZoeService} zoe
```

Comment:
```suggestion
 * @param {ERef<ZoeService>} zoe
```

Most (all?) functions that have a `zoe` param typed as `ZoeService` seem to only use it in ways compatible with it being a promise-for-ZoeService. I suspect we want to allow this. If so, those parameters should be typed `ERef<ZoeService>`. By contrast, the two `Instance` parameters below typically designate remote objects. But they are used for their identity directly and so must not be promises. Their typing should remain as is. 

If the intention is to disallow a promise for the `zoe` argument, then no change suggested.

Reply:
done